### PR TITLE
SDIT-1125 Force overrideAccess annotations to set a role

### PR DESCRIPTION
**Pull request by karenmillermoj**
This PR is just a refactor PR
Highlighting SYSTEM_USER usage directly will make it easier to remove it's usage.

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java [hunk 1]
```diff
@@ -55,11 +57,11 @@ public static boolean hasRoles(final String... allowedRoles) {
     }
 
     public boolean isOverrideRole(final String... overrideRoles) {
-        final var roles = overrideRoles.length > 0 ? getRoles(overrideRoles) : List.of("SYSTEM_USER");
-        return hasMatchingRole(roles, getAuthentication());
+        return hasMatchingRole(getRoles(overrideRoles), getAuthentication());
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java:60]: We can be safely removed as no overrideRole calls have empty role check

File: src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java [hunk 2]
```diff
@@ -55,11 +57,11 @@ public static boolean hasRoles(final String... allowedRoles) {
...
     }
 
     private static boolean hasMatchingRole(final List<String> roles, final Authentication authentication) {
+        if (roles.isEmpty()) log.error("Matching role check failed - no roles to match against");
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java:64]: Suggested as part of a previous PR

File: src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java [hunk 3]
```diff
@@ -55,6 +57,7 @@ public static boolean hasRoles(final String... allowedRoles) {
     }
 
     public boolean isOverrideRole(final String... overrideRoles) {
+        if (overrideRoles.length > 0) log.error("Matching role check failed - no roles to match against");
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java:60]: This will be removed in a later PR

Comment (mikehalmamoj) [reply src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java:60]: I don't understand - the if says there are roles but the error says there aren't

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java [hunk 4]
```diff
@@ -462,7 +462,7 @@ public List<Assessment> getAssessments(@PathVariable("bookingId") @Parameter(des
         @ApiResponse(responseCode = "404", description = "Requested resource not found.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))}),
         @ApiResponse(responseCode = "500", description = "Unrecoverable error occurred whilst processing request.", content = {@Content(mediaType = "application/json", schema = @Schema(implementation = ErrorResponse.class))})})
     @GetMapping("/{bookingId}/caseNotes")
-    @VerifyBookingAccess
+    @VerifyBookingAccess(overrideRoles = {"SYSTEM_USER"})
```
Comment (mikehalmamoj) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/BookingResource.java:465]: I thought the point was we're getting rid of SYSTEM_USER?

File: src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java (hunk 3 above)
Comment (karenmillermoj) [reply src/main/java/uk/gov/justice/hmpps/prison/security/AuthenticationFacade.java:60]: yes - that is wrong

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**